quint/src/parsing/ToIrListener.ts

```diff
@@ -220,89 +219,79 @@ export class ToIrListener implements QuintListener {
   /** **************** translate operator definititons **********************/
 
   // translate a top-level or nested operator definition
-  exitOperDef(ctx: p.OperDefContext) {
+  exitAnnotatedOperDef(ctx: p.AnnotatedOperDefContext) {
     const name = ctx.normalCallName().text
-    const [params, typeTag] = this.processOpDefParams(ctx)
-    // get the definition body
+
+    const params = ctx._annotOperParam
+      .map(_ => popOrFail(this.paramStack, 'violated grammar of annotated AnnotatedOperDef'))
```

Comment:
`popOrFail` already includes a "violated grammar" prefix, we can remove it from here for clarity
```suggestion
      .map(_ => popOrFail(this.paramStack, 'annotated AnnotatedOperDef'))
```